File: 은지/LV2/짝지어 제거하기.py
def solution(s):
    answer = 0
    stack = []
    for x in s:
        if len(stack):
            v = stack.pop(-1)
            if x != v:
                stack.append(v)
                stack.append(x)
        else:
            stack.append(x)
    if len(stack) == 0:
        answer = 1
    return answer

s = "baabaa"
print(solution(s))

File: 은지/LV2/짝지어 제거하기.js
function solution(s)
{
    var answer = 0;
    var stack = []
    for (let x of s) {
      if (stack.length) {
        var v = stack.pop()
        if (x !== v) {
          stack.push(v)
          stack.push(x)
        }
      } else {
        stack.push(x)
      }
    }
    if (stack.length === 0) {
      answer = 1
    }
    return answer;
}

s = "baabaa"
console.log(solution(s))